summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SpoutProvider.scala

```diff
@@ -62,95 +61,77 @@ case class SpoutProvider(builder: StormTopologyBuilder, node: StormNode) {
+      sourceParallelism,
+      spoutStormMetrics.metrics,
+      formattedSummerSpout,
       builder,
       maxEmitPerExecute,
-      keyValueShards,
       flushExecTimeCounter,
       executeTimeCounter
     )
   }
 
-  private def getCountersForJob: Seq[(Group, Name)] = JobCounters.getCountersForJob(builder.jobId).getOrElse(Nil)
-
   private def counter(nodeName: Group, counterName: Name): Counter with Incrementor =
     new Counter(Group("summingbird." + nodeName.getString), counterName)(builder.jobId) with Incrementor
 
-  // Tormenta bug when openHook might get lost if it doesn't happen right before the getSpout.
-  private def getStormSpout[T](spout: Spout[T]): IRichSpout = {
-    val register: Externalizer[(TopologyContext) => Unit] = createRegistrationFunction
-    spout.openHook(register.get).getSpout
-  }
-
   /**
    * This method creates the counters expicit for the KeyValueSpout.
    * The function to register all the counters is created with all the registered metrics.
    */
   private def getKeyValueSpout[K, V](
-    tormentaSpout: Spout[(Timestamp, (K, V))],
-    sNode: SummerNode[Storm]): IRichSpout = {
+    sourceParallelism: Int,
+    tormentaSpout: Spout[Item[(K, V)]],
+    sNode: SummerNode[Storm]
+  ): AggregatedSpout[K, V] = {
     val summerBuilder = BuildSummer(builder, node)
     val nodeName = builder.getNodeName(node)
     val flushExecTimeCounter = counter(Group(nodeName), Name("spoutFlushExecTime"))
     val executeTimeCounter = counter(Group(nodeName), Name("spoutEmitExecTime"))
-    createSpoutToFeedSummer[K, V](sNode, summerBuilder, tormentaSpout, flushExecTimeCounter, executeTimeCounter)
+    createSpoutToFeedSummer(sourceParallelism, sNode, summerBuilder, tormentaSpout, flushExecTimeCounter, executeTimeCounter)
   }
 
-  /**
-   * This function takes the metrics and counters to register to the TopologyContext of a JobID and outputs a function which can be called on openHook.
-   */
-  private def createRegistrationFunction: Externalizer[(TopologyContext) => Unit] = {
-    val counters: Seq[(Group, Name)] = getCountersForJob
-    Externalizer({ context: TopologyContext =>
-      // Register metrics passed in SpoutStormMetrics option.
-      spoutStormMetrics.metrics().foreach {
-        x: StormMetric[IMetric] =>
-          context.registerMetric(x.name, x.metric, x.interval.inSeconds)
-      }
-      // Register summingbird counter metrics.
-      StormStatProvider.registerMetrics(builder.jobId, context, counters)
-      SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
-    })
-  }
+  private def getRawSpout[T](
+    sourceParallelism: Int,
+    tormentaSpout: Spout[Item[T]]
+  ): ItemSpout[T] =
+    Topology.RawSpout(sourceParallelism, spoutStormMetrics.metrics, tormentaSpout)
 
   /**
    * This is the decision logic.
    * If the spout is followed by a summer wraps the spout in KeyValueSpout
    */
-  def apply: (Int, IRichSpout) = {
+  def apply[T]: Topology.Spout[_] = {
```

Comment:
This `[T]` represents type of events we emit from chain of producers we put on this `Spout`. 
For return value it can be both `ItemSpout[T]` and `AggregatedSpout[K, V]` if `T = (K, V)` (common type for them is just `Spout[_]`). 

I kept this parameter because later in code tormenta's spout has a type of `Spout[Item[T]]` regardless to what `T` is - either just `T` or `(K, V)`.